rasterio/warp.py

```diff
@@ -88,11 +88,8 @@ def transform_geom(
         Transformed geometry in GeoJSON dict format
     """
 
-    loose_gdal_version = filter(
-        lambda x: x.isdigit(),
-        rasterio.__gdal_version__.split('.'))
-    loose_gdal_version = tuple(map(int, loose_gdal_version))
-    if loose_gdal_version[:2] >= (2, 2) and not antimeridian_cutting:
+    if (not antimeridian_cutting and
+            GDALVersion.runtime().at_least('2.2')):
```

Comment:
I was trying to make this comparison as semantically meaningful as possible, so that it was more readable in context.